clippy_lints/src/utils/internal_lints.rs

```diff
@@ -395,3 +401,111 @@ fn is_trigger_fn(fn_kind: FnKind<'_>) -> bool {
         FnKind::Closure(..) => false,
     }
 }
+
+declare_lint_pass!(CollapsibleCalls => [COLLAPSIBLE_SPAN_LINT_CALLS]);
+
+impl  EarlyLintPass for CollapsibleCalls {
+    fn check_expr(&mut self, cx: &EarlyContext<'_>, expr: &ast::Expr) {
+        use ast::ExprKind;
+
+        if_chain! {
+            if let ExprKind::Call(ref func, ref and_then_args) = expr.kind;
+            if let ExprKind::Path(None, ref path) = func.kind;
+            if let match_path_ast(path, &["span_lint_and_then"]);
```

Comment:
Brute force: Remove lines until the error disappears :man_shrugging: 

```suggestion
            if match_path_ast(path, &["span_lint_and_then"]);
```